admin/election_controller.go

```diff
@@ -0,0 +1,89 @@
+package admin
+
+import (
+	"encoding/json"
+
+	"github.com/google/uuid"
+	"github.com/vocdoni/blind-csp/model"
+	"github.com/vocdoni/blind-csp/types"
+	"go.vocdoni.io/dvote/httprouter"
+	"go.vocdoni.io/dvote/httprouter/bearerstdapi"
+)
+
+type ElectionController struct {
+	store model.ElectionStore
+}
+
+func NewElectionController(store model.ElectionStore) *ElectionController {
+	return &ElectionController{store: store}
+}
+
+func (c *ElectionController) Create(msg *bearerstdapi.BearerStandardAPIdata, ctx *httprouter.HTTPContext) error {
+	newElection := model.Election{}
+	if err := json.Unmarshal(msg.Data, &newElection); err != nil {
+		return err
+	}
+
+	election, err := c.store.CreateElection(&newElection)
+	if err != nil {
+		return err
+	}
+
+	electionBearerToken := uuid.NewSHA1(uuid.NameSpaceDNS, []byte(election.ID.String()))
+	responseMap := map[string]interface{}{
+		"adminToken": electionBearerToken.String(),
+		"data":       election,
+	}
+
+	resp, err := json.MarshalIndent(responseMap, "", " ")
+	if err != nil {
+		return err
+	}
+
+	return ctx.Send(resp, bearerstdapi.HTTPstatusCodeOK)
+}
+
+func (c *ElectionController) Election(msg *bearerstdapi.BearerStandardAPIdata, ctx *httprouter.HTTPContext) error {
+	var electionID types.HexBytes
+	if err := electionID.FromString(ctx.URLParam("electionId")); err != nil {
+		return err
+	}
+	election, err := c.store.Election(electionID)
+	if err != nil {
+		return err
+	}
+	resp, err := json.MarshalIndent(election, "", " ")
+	if err != nil {
+		return err
+	}
+	return ctx.Send(resp, bearerstdapi.HTTPstatusCodeOK)
+}
+
+func (c *ElectionController) Delete(msg *bearerstdapi.BearerStandardAPIdata, ctx *httprouter.HTTPContext) error {
+	electionBearerToken := uuid.NewSHA1(uuid.NameSpaceDNS, []byte(ctx.URLParam("electionId"))).String()
+	if msg.AuthToken != electionBearerToken {
+		return ctx.Send([]byte(`{"ok": "false", "code": "401", "reason":"Authentication failed"}`), bearerstdapi.HTTPstatusCodeErr)
```

Comment:
There should be a type for the error messages. 

You could do something like the following, then you can call `ctx.Send(new(apiError).Set(401, "authentication failed").MustMarshal())`

```
type apiError struct {
  Ok bool `json:"ok"`
  ....
}

func (e* apiError) Set(code int, reason string) *apiError {
  e.code = code
  e.reason = reason
  e.ok = false
  return e
 }

func (e *apiError) MustMarshal() []byte {
  data, err := json.Marshal(e)
  if err != nil { panic(err) }
  return data
 }
```

Just an idea.

Reply:
Not really sure about it, but created the type ApiResponse to set an standard. Please, check it out

Reply:
Actually... changed it a bit... wait 'till commit...